core/src/main/java/de/danoeh/antennapod/core/receiver/MediaButtonReceiver.java

```diff
@@ -30,9 +31,23 @@ public void onReceive(Context context, Intent intent) {
 			Intent serviceIntent = new Intent(context, PlaybackService.class);
 			serviceIntent.putExtra(EXTRA_KEYCODE, event.getKeyCode());
 			serviceIntent.putExtra(EXTRA_SOURCE, event.getSource());
+			serviceIntent.putExtra(EXTRA_KEY_SOURCE, Source.fromIntent(intent));
+
 			ContextCompat.startForegroundService(context, serviceIntent);
 		}
 
 	}
 
+	public enum Source {
+		NOTIFICATION, WIDGET, MEDIA_BUTTON;
```

Comment:
I meant for it to refer to hardware buttons, chose that name based on the description of [Android's MediaButtonReceiver](https://developer.android.com/reference/androidx/media/session/MediaButtonReceiver). But that can be confusing because this class receives more than just hardware button events (on my phone, the widget actions also go through this path).

In this implementation it really refers to all other sources of **external** media events that did not come from AntennaPod's own widget or notification. It gets used later on in the playback service to determine if we should chose the default media action or if we should go to the user's hardware button preferences.

A more explicit name would be better. Maybe HARDWARE? HARDWARE_BUTTON?

Reply:
HARDWARE_BUTTON sounds good and a lot more clear